### PR TITLE
7d

### PR DESCRIPTION
**Pull request by gmerlo**
add support for 7d launch

### CODE REVIEW

**Comment by gmerlo** [comment]
One check fails because the cmake version is too old, the sycl one fails in the test but I don't know where to see the logs (if any) to see what happened. I don't have a sycl platform to test, so I may have easily missed something

File: include/gtensor/assign.h [hunk 1]
```diff
@@ -219,6 +244,24 @@ __global__ void kernel_assign_6(Elhs lhs, Erhs _rhs)
...
+  auto rhs = _rhs;
+  int tidx = threadIdx.x + blockIdx.x * blockDim.x;
+  int tidy = threadIdx.y + blockIdx.y * blockDim.y;
+  int tidz = blockIdx.z;
+  if (tidx < lhs.shape(0) * lhs.shape(1) &&
+      tidy < lhs.shape(2) * lhs.shape(3)) {
+    int j = tidx / lhs.shape(0), i = tidx % lhs.shape(0);
+    int l = tidy / lhs.shape(2), k = tidy % lhs.shape(2);
+    int m = tidz % lhs.shape(5) % lhs.shape(4);
+    int n = tidz % lhs.shape(5) / lhs.shape(4), int o = tidz / lhs.shape(5);
```
Comment (bd4) [inline include/gtensor/assign.h:259]: Should be semicolon and new line instead of comma.

**Review by bd4 — CHANGES_REQUESTED** [review]
Can you add a 7d assign test?

I am looking into the sycl and cmake failures, starting with forcing a CI run on current master branch in case it is not caused by this pr at all.

File: tests/test_launch.cxx [hunk 2]
```diff
@@ -242,4 +258,36 @@ TEST(gtensor, device_launch_6d)
...
+  gt::gtensor<double, 7> h_b(h_a.shape());
+  gt::gtensor<double, 7> h_b_expected(h_a.shape());
+
+  for (int i = 0; i < h_a.shape(0); i++) {
+    for (int j = 0; j < h_a.shape(1); j++) {
+      for (int k = 0; k < h_a.shape(2); k++) {
+        for (int l = 0; l < h_a.shape(3); l++) {
+          for (int m = 0; m < h_a.shape(4); m++) {
+            for (int n = 0; n < h_a.shape(5); n++) {
+              for (int o = 0; n < h_a.shape(6); o++) {
```
Comment (bd4) [inline tests/test_launch.cxx:274]:
```suggestion
              for (int o = 0; o < h_a.shape(6); o++) {
```

**Review by bd4 — CHANGES_REQUESTED** [review]

**Review by bd4 — APPROVED** [review]

**Pull request merged**